### DISCUSSION TITLE
Tailwind CSS or CSS classes does not working properly while creating elements using vanilla js

### DISCUSSION THREAD

**Discussion by violetbee** [tailwindlabs/tailwindcss#13570] [Help]
**What version of Tailwind CSS are you using?**

v3.4.1

**What build tool (or framework if it abstracts the build tool) are you using?**

postcss-cli 8.4.35, vite 5.1.4, vanilla js

**What version of Node.js are you using?**

 v18.12.0

**What browser are you using?**

Chrome

**What operating system are you using?**

Windows

**Reproduction URL**
https://github.com/violetbee/tw-vite-issue

**Describe your issue**

I have created two different classes, one is based on tw and the other one is classic css. I added these classes to element while creating but it didnt work. 

![image](https://github.com/tailwindlabs/tailwindcss/assets/61601796/857782c9-4f91-496e-8978-0b31b87ef0d0)
![image](https://github.com/tailwindlabs/tailwindcss/assets/61601796/570eb9ed-2ef6-407f-95eb-a801af39503c)
![image](https://github.com/tailwindlabs/tailwindcss/assets/61601796/7a455af7-90f9-4ccc-a4cc-d0de36ba8a11)
![image](https://github.com/tailwindlabs/tailwindcss/assets/61601796/69238710-bef2-4f1f-a9e0-f4d60fa35f84)

I can only assign tailwind CSS classes to the elements that I have created with Javascript. Why?

**Comment by wongjn** [comment]
It seems like the PostCSS plugin you have, `postcss-uncss`, is removing the rules from the CSS.

As per [the documentation for `postcss-uncss`](https://www.npmjs.com/package/postcss-uncss#options), you could:
- Use the `ignore` configuration:
  ```diff
   'postcss-uncss': {
     html: ['./src/index.html'],
  +  ignore: ['.base-css-class', '.tw-apply-class'],
   },
  ```
- Or use an inline comment:
  ```css
  /* uncss:ignore */
  .base-css-class {
    …
  }
  /* uncss:ignore */
  .tw-apply-class {
    …
  }
  ```

Be sure to restart the Vite development server after either change.
